src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1202,43 +1156,38 @@ public List<String> getInactiveConsumerIds(Instant lastCheckedInRetention, Insta
      *     Org namespace to search
      * @return Consumer that matches the given
      */
-    @SuppressWarnings("unchecked")
-    @Transactional
-    public CandlepinQuery<Consumer> getHypervisorsBulk(Iterable<String> hypervisorIds, String ownerId) {
+    public List<Consumer> getHypervisorsBulk(Collection<String> hypervisorIds, String ownerId) {
         if (hypervisorIds == null || !hypervisorIds.iterator().hasNext()) {
-            return this.cpQueryFactory.<Consumer>buildQuery();
+            return new ArrayList<>();
         }
 
-        DetachedCriteria criteria = DetachedCriteria.forClass(Consumer.class)
-            .createAlias("hypervisorId", "hvsr")
-            .add(Restrictions.eq("ownerId", ownerId))
-            .add(this.getHypervisorIdRestriction(hypervisorIds))
-            .addOrder(org.hibernate.criterion.Order.asc("hvsr.hypervisorId"))
-            .setFetchMode("type", FetchMode.SELECT);
-
-        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria)
-            .setLockMode(LockModeType.PESSIMISTIC_WRITE);
-    }
+        List<String> lowerCaseHypervisorIds = hypervisorIds.stream()
+            .map(String::toLowerCase)
+            .toList();
 
-    private Criterion getHypervisorIdRestriction(Iterable<String> hypervisorIds) {
-        Disjunction disjunction = Restrictions.disjunction();
-        for (String hid : hypervisorIds) {
-            disjunction.add(Restrictions.eq("hvsr.hypervisorId", hid.toLowerCase()));
-        }
+        String jpql = """
+            SELECT c FROM Consumer c
+            JOIN c.hypervisorId h
+            WHERE c.ownerId = :ownerId
+                AND LOWER(h.hypervisorId) IN :lowerCaseHypervisorIds
+            ORDER BY h.hypervisorId ASC""";
 
-        return disjunction;
+        return getEntityManager().createQuery(jpql, Consumer.class)
+            .setParameter("ownerId", ownerId)
+            .setParameter("lowerCaseHypervisorIds", lowerCaseHypervisorIds)
```

Comment:
This could run into the limit of parameters on the IN operator. The input needs to be properly partitioned.

Reply:
done